ConsumerApi.Sdk/Configuration.cs

```diff
@@ -16,4 +16,15 @@ public class AuthenticationConfiguration
         public required string Username { get; init; }
         public required string Password { get; init; }
     }
+
+    public object CloneWith(AuthenticationConfiguration newAuthentication)
+    {
+        return new Configuration()
+        {
+            Authentication = newAuthentication,
+            JsonSerializerOptions = JsonSerializerOptions,
+            BaseUrl = null!,
+            ApiVersion = null!
+        };
+    }
```

Comment:
I don't understand this method. It says "clone" but out of the 4 properties, only 1 is "cloned", while 2 are set to null (even though the properties are not even allowed to be null), and one is overridden.